Trakt/Helpers/UserDataManagerEventsHelper.cs

```diff
@@ -109,29 +114,56 @@ public void ProcessUserDataSaveEventArgs(UserDataSaveEventArgs userDataSaveEvent
             }
 
             // If it's not the series we're currently storing, upload our episodes and reset the arrays
-            if (!userPackage.CurrentSeriesId.Equals(episode.Series.Id))
+            if (userPackage.CurrentSeriesId != null)
             {
-                if (userPackage.SeenEpisodes.Any())
+                if (!userPackage.CurrentSeriesId.Equals(episode.Series.Id))
                 {
-                    _traktApi.SendEpisodePlaystateUpdates(
-                        userPackage.SeenEpisodes.ToList(),
-                        userPackage.TraktUser,
-                        true,
-                        CancellationToken.None).ConfigureAwait(false);
-                    userPackage.SeenEpisodes.Clear();
-                }
+                    if (userPackage.SeenEpisodes.Any())
+                    {
+                        _traktApi.SendEpisodePlaystateUpdates(
+                            userPackage.SeenEpisodes.ToList(),
+                            userPackage.TraktUser,
+                            true,
+                            CancellationToken.None).ConfigureAwait(false);
+                        userPackage.SeenEpisodes.Clear();
+                    }
 
-                if (userPackage.UnSeenEpisodes.Any())
-                {
-                    _traktApi.SendEpisodePlaystateUpdates(
-                        userPackage.UnSeenEpisodes.ToList(),
-                        userPackage.TraktUser,
-                        false,
-                        CancellationToken.None).ConfigureAwait(false);
-                    userPackage.UnSeenEpisodes.Clear();
+                    if (userPackage.UnSeenEpisodes.Any())
+                    {
+                        _traktApi.SendEpisodePlaystateUpdates(
+                            userPackage.UnSeenEpisodes.ToList(),
+                            userPackage.TraktUser,
+                            false,
+                            CancellationToken.None).ConfigureAwait(false);
+                        userPackage.UnSeenEpisodes.Clear();
+                    }
+
+                    userPackage.CurrentSeriesId = episode.Series.Id;
                 }
+                else
```

Comment:
What is the difference between the if and the else?

Reply:
The if triggers when we change the series, the else triggers when we are in the same series and the current queue exceeds 100 items.

Reply:
You can more or less remove the if-else by saving the MaxCount based on `userPackage.CurrentSeriesId != null`